### ISSUE TITLE
BackupFilesExclude values not being shown in server/host config web UI

### ISSUE THREAD

**Issue by JacksonTech** [backuppc/backuppc#112]
I have some server-wide and host-wide (override) BackupFilesExclude keys set for my rsync xfer method. In BackupPC 3.x, these values were shown in the web UI. In 4.1.2, none of the values are shown.

To be more specific, the "override" box is checked on the per-host overrides, but only the "*" share is shown, and it has no values. (I can hit "add" and the boxes will appear, but there is nothing in them.)

Looking in the page source, it looks like they're being inserted into the page, but the form is not displaying them:

```
<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_0" value="/proc">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_1" value="/sys">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_10" value="/tmp">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_11" value="/var/tmp">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_12" value="/var/run">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_13" value="/var/lock">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_14" value="/var/cache/apt/archives">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_15" value="/dev/*">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_16" value="/home/*/mnt">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_17" value="/home/*/.ccache">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_2" value="/media">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_3" value="/mnt">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_4" value="/home/*/.cache">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_5" value="/home/*/.thumbnails">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_6" value="/home/*/.mozilla/firefox/*.default/Cache">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_7" value="/home/*/.mozilla/firefox/*.default/OfflineCache">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_8" value="/home/*/.local/share/Trash">

<input type="hidden" name="orig_zZ_BackupFilesExclude_zZ_0_zZ_9" value="/home/*/.gvfs/">
```

**Comment by craigbarratt** [comment]
Cody,

Hmm, I fixed one bug related to SCGI with these symptoms and it was
included in  4.1.2.  Are you using SCGI?  If so can you try to regular CGI
interface and confirm it works?

Thanks,
Craig

**Comment by JacksonTech** [comment]
@craigbarratt Hmm, I'm using CGI (probably not wise, but I'm the only guy on the server) through the Ubuntu apache-suexec-custom package and Apache. 

I can try it on a VM (both SCGI and CGI) using my existing config files. I'll get back to you in a few days.

**Comment by mntbighker** [comment]
I installed (upgraded) 4.1.2 today with the epel COPR. I'm not using SCGI and I have the same problem. I have tried all different manual methods of setting the BackupFilesExclude, and tried using the GUI. You can set them ONCE in the GUI, and see the change reflected in config.pl. But once you navigate away from Xfer and make ANY change that rewrites config.pl, you will have an empty Exclude. If you edit the Exclude manually, the GUI will fail to display the settings, and any other change will erase them all. The GUI just keeps resetting them to be empty. Even if you are looking at another settings panel. It's as if the issue is that the GUI can't parse that piece of the config.pl correctly, so it just displays it empty. So any save resets it to empty. The LOG file and the httpd logs seem to offer no clue either. The Exclude change at 16:29:05 below wasn't me, but it says it was. As you can see, I made a different change and it reset it.

2017-05-11 16:28:01 tsadmin changed BackupFilesExclude in main config to {'*' => ['lastlog','lost+found']} from {'*' => []}
2017-05-11 16:28:01 Reloading config/host files via CGI request
2017-05-11 16:28:01 Next wakeup is 2017-05-11 17:00:00
2017-05-11 16:29:05 tsadmin changed BackupFilesExclude in main config to {'*' => []} from {'*' => ['lastlog','lost+found']}
2017-05-11 16:29:05 tsadmin changed MaxOldPerPCLogFiles in main config to '12' from '10'

**Comment by craigbarratt** [comment]
This was a bug that I fixed in 4.1.2, but for some reason it is still
happening.  It is specific to SCGI.  Can you confirm you are using SCGI?

**Comment by craigbarratt** [comment]
While it works on my 4.1.2 setup, I have been able to replicate the problem.  It only happens if the main config.pl BackupFilesExclude/Only settings are non-empty. (On my setup they are emtpy.) . I'm working on a fix.

For the logo link, it got changed by a PR (968534a), and yes it's wrong.  I'll change it to point it at the new project home page.

**Comment by craigbarratt** [comment]
I pushed a potential fix (7936184).  It's only very lightly tested.

**Comment by mntbighker** [comment]
I'm curious since the setting syntax didn't change, why the version 3.x parser worked and it's broken in 4.x? I am NOT on SCGI either.

Thanks for the quick response though. I really appreciate it.

**Comment by craigbarratt** [comment]
Mark,

SCGI doesn't parse input variables the same way as CGI.  So it was broken
in SCGI.  My "fix" in 4.1.2 fixed that, but broke the case where the main
config.pl entries were non-empty (for both SCGI and CGI).

I pushed a fix
<https://github.com/backuppc/backuppc/commit/7936184a9ec049fef3d0d67e012b23d79eb336f1>
last night that you can try out.

Craig

**Comment by mntbighker** [comment]
That seems to work both in "Edit Config" and inherit properly in client config edit.

Nicely done, hope it doesn't bork anything else ;-)

**Comment by gillescoi** [comment]
hi,
i try your commit <7936184> on the EditConfig.pm on my 4.1.2 install with SCGI enabled.
The BackupFilesExclude continue to be inconsistent in the per host web interface. My host config look like this :
$Conf{BackupFilesExclude} = {
  '/usr' => [
    '/lost+found'
  ],
  '/var' => [
    '/lock',
    '/run',
    '/backups',
    '/cache',
    '/lost+found',
    '/lib/backuppc'
  ],
  '/srv/apache' => [
    '/lost+found',
    '/var/lib/backuppc'
  ],
  '/jail' => [
    '/lost+found'
  ],
  '/home' => [
    '/lost+found'
  ],
  '/scan' => [
    '/lost+found'
  ],
  '/opt' => [
    '/lost+found'
  ]
};
show like this in the web view (after any save this will be write in the host conf file) : 
$Conf{BackupFilesExclude} = {
  '/var' => [
    '/lost+found'
  ]
};

best regards

**Comment by craigbarratt** [comment]
Oops, you're right.  SCGI has yet more incompatibilities with CGI.  I pushed another change that should fix this issue (d9939a7).

**Comment by gillescoi** [comment]
hi,
works perfectly with your last commit, great job !.
thanks a lot, have a nice day
